### PR TITLE
Copter: Set the landing waiting time at fail safe to the set value

### PR DESCRIPTION
**Pull request by muramura**
I don't know why failsafe is fixed at 4 seconds.
I think it would be better to be able to change 4 seconds.

### CODE REVIEW

**Comment by muramura** [comment]
@khancyr san. Comment Thanks.

I live in Japan. In Japan, telemetry information cannot be viewed on the radio.
I don't know why to wait 4 seconds in failsafe.
If there is a process that takes 4 seconds after detecting fail safe, I think that it is good as it is.

In Japan, the radio cannot receive telemetry by itself.
Many telemetry messages of ArduPilot are useless in Japan.
In Japan, alarms, LED lights, blinks, and colors are often used.

I am fail safe and want to decide whether to wait 4 seconds or immediately.

File: ArduCopter/mode_land.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ void ModeLand::gps_run()
         motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::THROTTLE_UNLIMITED);
 
         // pause before beginning land descent
-        if(land_pause && millis()-land_start_time >= LAND_WITH_DELAY_MS) {
+        if(land_pause && millis()-land_start_time >= static_cast<uint32_t>(copter.g.land_delay_time)) {
```
Comment (peterbarker) [inline ArduCopter/mode_land.cpp:77]: This isn't a `uint32_t`.  Our pattern is usually `.get()`.

Similarly elsewhere.

File: ArduCopter/Parameters.cpp [hunk 2]
```diff
@@ -723,6 +723,15 @@ const AP_Param::Info Copter::var_info[] = {
     // @Path: ../libraries/AP_Vehicle/AP_Vehicle.cpp
     { AP_PARAM_GROUP, "", Parameters::k_param_vehicle, (const void *)&copter, {group_info : AP_Vehicle::var_info} },
 
+    // @Param: LAND_DELAY_TIME
+    // @DisplayName: Waiting time to land
```
Comment (peterbarker) [inline ArduCopter/Parameters.cpp:727]:
```suggestion
    // @DisplayName: Land Pause Delay
```

File: ArduCopter/Parameters.cpp [hunk 3]
```diff
@@ -723,6 +723,15 @@ const AP_Param::Info Copter::var_info[] = {
...
+    // @Description: Waiting time to land
```
Comment (peterbarker) [inline ArduCopter/Parameters.cpp:728]:
```suggestion
    // @Description: When switching to land mode due to a failsafe the vehicle will pause for this many milliseconds before starting to descend
```

File: ArduCopter/Parameters.cpp [hunk 4]
```diff
@@ -723,6 +723,15 @@ const AP_Param::Info Copter::var_info[] = {
...
+    // @Range: 0 30000
+    // @Increment: 1
+    // @Units: ms
+    // @User: Standard
+    GSCALAR(land_delay_time, "LAND_DELAY_TIME", LAND_WITH_DELAY_MS),
```
Comment (peterbarker) [inline ArduCopter/Parameters.cpp:733]: We may want 
```suggestion
    GSCALAR(land_delay_time, "LAND_DELAY_MS", LAND_WITH_DELAY_MS),
```

**Review by peterbarker — CHANGES_REQUESTED** [review]

**Comment by rmackay9** [comment]
I'm not sure that this is really worth the cost of a new parameter.  Each new parameter takes up some flash and more importantly adds complexity for the user.  I think if the delay is too long or too short you can perhaps adds a #define into APM_Config.h.  Txs anyway

**Pull request closed without merging**